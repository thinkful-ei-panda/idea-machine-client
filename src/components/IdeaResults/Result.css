li{
  list-style-type: none;
  background-color: #f5f5f6;
  text-align: left;
  padding:8px;
  margin: 15px;
  display:flex;
  flex-direction: column;
  box-shadow: 2px 2px 4px #808e95;
  word-wrap: break-word;
  width:320px;
  align-self: flex-start;
}

.buttonsContainer{
  display:flex;
  flex-direction: row;
  padding-top: 20px;
  padding-bottom: 20px;  
}

.buttonsContainer div{
  flex: 1;
  text-align: center;
}

button{
  height: 35px;
  width: 95px;
  box-shadow: 2px 2px 4px #808e95;
}

.secondary{
  background-color: #b0bec5;
}

h3{
  margin: 0;
}

section{
  padding-top: 10px;
}