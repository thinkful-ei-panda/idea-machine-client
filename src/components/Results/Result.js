import React from 'react'

const Result = (props) => {
  return (
    <div>
      {props.result}
    </div>
  )
}

export default Result