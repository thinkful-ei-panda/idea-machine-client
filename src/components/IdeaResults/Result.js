import React from 'react'
import {useLocation} from 'react-router-dom'
import TokenService from '../../services/token-service';
import './Result.css';

const Result = (props) => {
    
  const {title,content,user_name,id,public_status,followed} = props.result

  let location = useLocation();

  return (
    <li id={id} className='idea'>
      <header>
        <h3>{title}</h3>
        {(location.pathname === '/' || location.pathname === '/tracked-ideas') && <div>by {user_name}</div>}
      </header>
      <section>
        {content}
      </section>
      <div className='buttonsContainer'>
      {(location.pathname === '/my-ideas' && public_status === false) && <div><button onClick = {e => props.handleMakePublicClick(e)} >Make public</button></div>}
      {(location.pathname === '/my-ideas' && public_status === true) && <div><button onClick = {e => props.handleMakePrivateClick(e)} >Make private</button></div>}
      {location.pathname === '/my-ideas' && <div><button onClick = {e => props.handleEditClick(e,props.history,title,content) } className='secondary'>Edit</button></div>}
      {location.pathname === '/my-ideas' && <div><button onClick = {e => props.handleDeleteClick(e)} className='secondary'>Delete</button></div>}
      {(location.pathname === '/' && TokenService.hasAuthToken() && !followed) && <div><button onClick = {e => props.handleFollowClick(e)}>Follow</button></div>}
      {(location.pathname === '/' && TokenService.hasAuthToken() && followed) && <div>Followed</div>}
      {location.pathname === '/tracked-ideas' && <div><button onClick = {e =>props.handleUnfollowClick(e)}>Unfollow</button></div>}
      </div>
    </li>
  )
}  

export default Result