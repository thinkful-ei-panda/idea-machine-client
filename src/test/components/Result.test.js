import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Result from '../../components/IdeaResults/Result';
describe(`Result Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <Result result={{title:'asdf',content:'aldksjf',user_name:'ladkjs',id:1,public_status:true,followed:'aldsfj'}}/>
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <Result result={{title:'asdf',content:'aldksjf',user_name:'ladkjs',id:1,public_status:true,followed:'aldsfj'}}/>
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})