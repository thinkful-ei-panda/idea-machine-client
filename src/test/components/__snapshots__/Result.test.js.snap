// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Component Snapshot test Renders the UI as expected 1`] = `
<li
  className="idea"
  id={1}
>
  <header>
    <h3>
      asdf
    </h3>
    <div>
      by 
      ladkjs
    </div>
  </header>
  <section>
    aldksjf
  </section>
  <div
    className="buttonsContainer"
  />
</li>
`;
